### PR TITLE
Consistent and improved naming for items built by emscripten ports

### PR DESCRIPTION
**Pull request by sbc100**
- Use .a when the library is an archive and .bc only when its bitcode
- always prefer .a archives over combined .o files when
  building with WASM_OBJECT_FILES=1.
- make library names consistent with upstream library names.
- Fix the directory names used by some ports so that they get cleared correctly by `clear_project_build`

### CODE REVIEW

File: tools/ports/sdl2.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ def get(ports, settings, shared):
 
   # get the port
   ports.fetch_project('sdl2', 'https://github.com/emscripten-ports/SDL2/archive/' + TAG + '.zip', SUBDIR)
+  libname = ports.get_lib_name('libSDL2')
```
Comment (kripken) [inline tools/ports/sdl2.py]: why not lowercase?

File: tools/system_libs.py [hunk 2]
```diff
@@ -800,12 +806,18 @@ def build_port(src_path, output_path, includes=[], flags=[], exclude_files=[], e
       objects.append(obj)
 
     run_commands(commands)
+    print('create_lib', output_path)
```
Comment (kripken) [inline tools/system_libs.py]: Unintended debug statement?

File: tools/ports/sdl2.py (hunk 1 above)
Comment (sbc100) [reply tools/ports/sdl2.py]: I'm just following the naming conversions normally used by SDL:
```
$ dpkg -L libsdl2-2.0-0 libsdl2-ttf-2.0-0 | grep .so
/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0.8.0
/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0
/usr/lib/x86_64-linux-gnu/libSDL2_ttf-2.0.so.0.14.0
/usr/lib/x86_64-linux-gnu/libSDL2_ttf-2.0.so.0
```
Normally when one want to link SDL on UNIX one would use `-lSDL` which corresponds the a library called `libSDL.{a,so}`.   This is also what SDLs build system calls the library.

Comment (kripken) [reply tools/ports/sdl2.py]: Oh, I see. Yeah, sounds good.

**Pull request merged**